DicomUltrasoundPlugin/DicomUltrasoundPlugin.py

```diff
@@ -565,10 +565,23 @@ def load(self,loadable):
     else:
       loadedNode = self.loadPhilips4DUSAsSequence(loadable)
 
+    # Save DICOM SOP instance UID into the sequence so DICOM metadata can be retrieved later if needed
+    if loadedNode:
+      loadedNode.SetAttribute('DICOM.instanceUIDs', slicer.dicomDatabase.instanceForFile(loadable.files[0]))
+
     # Show sequence browser toolbar if a sequence has been loaded
     if loadedNode and loadedNode.IsA('vtkMRMLSequenceNode'):
       sequenceBrowserNode = slicer.modules.sequences.logic().GetFirstBrowserNodeForSequenceNode(loadedNode)
       if sequenceBrowserNode:
+
+        # Save DICOM SOP instance UID into the master node so DICOM metadata can be retrieved later from
+        # the proxy nodes if needed
+        masterSequenceNode = sequenceBrowserNode.GetMasterSequenceNode()
+        masterSequenceNode.SetAttribute('DICOM.instanceUIDs', loadedNode.GetAttribute('DICOM.instanceUIDs'))
+
+        volumeNode = browserNode.GetProxyNode(loadedNode)
+        volumeNode.SetAttribute('DICOM.instanceUIDs', loadedNode.GetAttribute('DICOM.instanceUIDs'))

```

Comment:
To follow-up on the commit originally posted by @lassoan in https://github.com/SlicerHeart/SlicerHeart/pull/28#discussion_r1264190428:

> The sequence browser node can browse many different images, so in general the DICOM instance UIDs cannot be saved into that.
> 
> Instead, the attribute need to be set on the volume node in the sequence (and those attributes are copied to the proxy node).

I am not clear which approach was implied ? 
* Getting the master node
* Or getting the proxy node

Reply:
You need to set the attribute of the data node that you set in the sequence. 

Content of the data node in the sequence (including the attributes) are copied to the proxy node during replay.